pkg/storage/etcd/etcd_helper.go

```diff
@@ -56,26 +61,59 @@ func (c *EtcdConfig) GetType() string {
 
 // implements storage.Config
 func (c *EtcdConfig) NewStorage() (storage.Interface, error) {
-	cfg := etcd.Config{
-		Endpoints: c.ServerList,
-		// TODO: Determine if transport needs optimization
-		Transport: &http.Transport{
-			Proxy: http.ProxyFromEnvironment,
-			Dial: (&net.Dialer{
-				Timeout:   30 * time.Second,
-				KeepAlive: 30 * time.Second,
-			}).Dial,
-			TLSHandshakeTimeout: 10 * time.Second,
-			MaxIdleConnsPerHost: 500,
-		},
-	}
-	etcdClient, err := etcd.New(cfg)
+	etcdClient, err := etcd.New(c.getEtcdClientConfig())
 	if err != nil {
 		return nil, err
 	}
 	return NewEtcdStorage(etcdClient, c.Codec, c.Prefix, c.Quorum), nil
 }
 
+func (c *EtcdConfig) getEtcdClientConfig() etcd.Config {
+	return etcd.Config{
+		Endpoints: c.ServerList,
+		Transport: c.newHttpTransport(),
+	}
+}
+
+func (c *EtcdConfig) newHttpTransport() etcd.CancelableTransport {
```

Comment:
A big TODO for a test. I'm not sure the best way. I'm assuming you manually tried it and it works? :)